client/blocks/plan-thank-you-card/index.jsx

```diff
@@ -19,57 +19,94 @@ import ThankYouCard from 'components/thank-you-card';
 import PlanIcon from 'components/plans/plan-icon';
 import { getPlanClass } from 'lib/plans/constants';
 
-const PlanThankYouCard = ( {
-	plan,
-	translate,
-	siteId,
-	siteUrl,
-	action
-} ) => {
-	const name = plan && translate( '%(planName)s Plan', {
-		args: { planName: getPlan( plan.productSlug ).getTitle() }
-	} );
-	const price = plan && formatCurrency( plan.rawPrice, plan.currencyCode );
-	const productSlug = plan && plan.productSlug;
-	const planClass = productSlug
-		? getPlanClass( productSlug )
-		: '';
-	const planIcon = productSlug
-		? <PlanIcon plan={ productSlug } />
-		: null;
-	const renderAction = () => {
+class PlanThankYouCard extends Component {
+	getPlanClass() {
+		const { plan } = this.props;
+		if ( ! plan || ! plan.productSlug ) {
+			return '';
+		}
+
+		return getPlanClass( plan.productSlug );
+	}
+
+	renderPlanName() {
+		const { plan, translate } = this.props;
+		if ( ! plan ) {
+			return '';
+		}
+
+		return translate( '%(planName)s Plan', {
+			args: { planName: getPlan( plan.productSlug ).getTitle() }
+		} );
+	}
+
+	renderPlanPrice() {
+		const { plan } = this.props;
+		if ( ! plan || ! plan.rawPrice || ! plan.currencyCode ) {
+			return '';
+		}
+
+		return formatCurrency( plan.rawPrice, plan.currencyCode );
+	}
+
+	renderPlanIcon() {
+		const { plan } = this.props;
+		if ( ! plan || ! plan.productSlug ) {
+			return null;
+		}
+
+		return <PlanIcon plan={ plan.productSlug } />;
+	}
+
+	renderAction() {
+		const { action } = this.props;
 		if ( action ) {
 			return action;
 		}
 
 		return null;
-	};
+	}
 
-	return (
-		<div className={ classnames( 'plan-thank-you-card', planClass ) }>
-			<QuerySites siteId={ siteId } />
-			<QuerySitePlans siteId={ siteId } />
-
-			<ThankYouCard
-				name={ name }
-				price={ price }
-				heading={ translate( 'Thank you for your purchase!' ) }
-				description={ translate( "Now that we've taken care of the plan, it's time to see your new site." ) }
-				buttonUrl={ siteUrl }
-				buttonText={ translate( 'Visit Your Site' ) }
-				icon={ planIcon }
-				action={ renderAction() }
-			/>
-		</div>
-	);
-};
+	renderDescription() {
+		const { description, translate } = this.props;
+		if ( description ) {
+			return description;
+		}
+
+		return translate(
+			"Now that we've taken care of the plan, it's time to see your new site."
```

Comment:
The plans thank you card currently is only shown to new WP.com users. We override this wording for the Jetpack use case though.